services/rsources/handler.go

```diff
@@ -404,3 +406,43 @@ func sqlFilters(jobRunId string, filter JobFilter) (fragment string, params []in
 	}
 	return filters, filterParams
 }
+
+func (sh *sourcesHandler) Monitor(ctx context.Context, lagGauge, replicationSlotGauge Gauger) {
+	if sh.sharedDB == nil {
+		sh.log.Warn("shared database is not configured, skipping logical replication monitoring")
+		return
+	}
+	logicalReplicationTrigger := func() <-chan time.Time {
+		return time.After(config.GetDuration("Rsources.stats.monitoringInterval", 10, time.Second))
+	}
+	go func() {
```

Comment:
why do we really need to put this under a `go routine`?

Reply:
we don't